file-formats/dsfi/type/zif_aff_dsfi_v1.intf.abap

```diff
@@ -2,18 +2,35 @@ INTERFACE zif_aff_dsfi_v1
   PUBLIC.
 
 
-  "! <p class="shorttext">Engine</p>
-  "! Engine
-  "! $values {@link zif_aff_dsfi_v1.data:co_engine}
-  TYPES: ty_engine TYPE zif_aff_types_v1=>ty_object_name_30.
+  TYPES:
+    "! <p class="shorttext">Engine</p>
+    "! Engine
+    "! $values {@link zif_aff_dsfi_v1.data:co_engine}
+    "! $default {@link zif_aff_dsfi_v1.data:co_engine.analytical_engine}
+    ty_engine TYPE zif_aff_types_v1=>ty_object_name_30.
 
   CONSTANTS:
     BEGIN OF co_engine,
       "! <p class="shorttext">Analytical Engine</p>
       "! Analytical engine
       analytical_engine TYPE ty_engine VALUE 'ANA',
+      "! <p class="shorttext">SQL Engine</p>
+      "! SQL engine
+      sql_engine TYPE ty_engine VALUE 'SQL',
     END OF co_engine.
 
+  TYPES:
+    BEGIN OF ty_sql_properties,
+      "! <p class="shorttext">AMDP Reference</p>
+      "! The name of the AMDP implementation reference
+      "! $required
+      amdp_reference TYPE c LENGTH 100,
+      "! <p class="shorttext">Auto Exposure</p>
+      "! Flag whether or not the SQL implementation of the
+      "! CDS scalar function will be automatically exposed in all SQL services
+      auto_exposure  TYPE abap_bool,
```

Comment:
The flag is only applicable to scalar function implementations done with a SQL engine. For this case, indeed the flag will be true for most cases. For analytical engine scalar functions however the flag will always be false, so it should be able to be omitted.
If there's a way to show always for SQL functions and never for analytical functions, that would be best. Otherwise it's better to keep the current setup.

Reply:
Internally, we treat $showAlways in the same way as $required. So it should be possible.

Reply:
Thank you, I've set the field to $showAlways